src/app/add-clients/helperFunctions.tsx

```diff
@@ -0,0 +1,254 @@
+};
+export const numberRegex = /^\d+$/;
+
+export const getNumberAdults = (
+    fieldSetter: SetFieldType,
+    errorSetter: SetErrorType,
+    adults: People[],
+    personType: PersonType
+): OnChangeType => {
+    return (event) => {
+        const input = event.target.value;
+        const numberPattern = /^\d+$/;
+        const newValue = adults;
+        const personIndex = newValue.findIndex((object) => object.personType === personType);
+        if (input === "") {
+            newValue[personIndex].quantity = 0;
+        } else if (input.match(numberPattern)) {
+            newValue[personIndex].quantity = parseInt(input);
+        } else {
+            newValue[personIndex].quantity = -1;
+        }
+        fieldSetter("adults", newValue);
+
+        const invalidAdultEntry = newValue.filter((value) => value.quantity === -1);
+        const nonZeroAdultEntry = newValue.filter((value) => value.quantity! > 0);
+        let newMessage = "";
+        if (invalidAdultEntry.length > 0) {
+            newMessage = "Please enter a valid entry.";
+        } else if (nonZeroAdultEntry.length === 0) {
+            newMessage = "This is a required field.";
+        }
+        errorSetter("adults", newMessage);
+    };
+};
+
+export const getChild = (
+    fieldSetter: SetFieldType,
+    children: People[],
+    index: number,
+    subFieldName: "personType" | "age"
+): OnChangeType => {
+    return (event) => {
+        const input = event.target.value;
+        if (subFieldName === "personType") {
+            children[index][subFieldName] = (
+                input !== "don't know" ? input : "child"
+            ) as PersonType;
+        } else {
+            children[index][subFieldName] = input !== "don't know" ? parseInt(input) : null;
+        }
+        fieldSetter("children", [...children]);
+    };
+};
+
+export const getBaby = (fieldSetter: SetFieldType, errorSetter: SetErrorType): OnChangeType => {
+    return (event) => {
+        const input = event.target.value;
+        if (input === "Yes") {
+            errorSetter("babyProducts", "N/A");
+            fieldSetter("babyProducts", true);
+        } else {
+            errorSetter("babyProducts", "");
+            if (input === "No") {
+                fieldSetter("babyProducts", false);
+            } else {
+                fieldSetter("babyProducts", null);
+            }
+        }
+    };
+};
+
+export const checkErrorOnSubmit = (
+    errorMessages: ErrorMessages,
+    errorSetter: React.Dispatch<React.SetStateAction<ErrorMessages>>
+): boolean => {
+    let errorExists = false;
+    let amendedErrorMessages = { ...errorMessages };
+    for (const [errorKey, errorMessage] of Object.entries(errorMessages)) {
+        if (errorMessage !== "") {
+            errorExists = true;
+        }
+        if (errorMessage === "N/A") {
+            amendedErrorMessages = {
+                ...amendedErrorMessages,
+                [errorKey]: "This is a required field.",
+            };
```

Comment:
same issue with special value for errors. Look out for these equality checks throughout the codebase so you don't end up with surprise errors when/if you refactor.